src/v/redpanda/admin_server.cc

```diff
@@ -531,8 +540,9 @@ ss::future<> admin_server::throw_on_error(
 }
 
 void admin_server::register_config_routes() {
-    static ss::httpd::handle_function get_config_handler =
-      []([[maybe_unused]] ss::const_req req, ss::reply& reply) {
+    thread_local static ss::httpd::handle_function get_config_handler =
```

Comment:
I don't see any benefit to making this handler thread_local nor static since there isn't any state maintained. We'd use a slot in the thread_local index and static has initialization guards. Am I missing something?

Reply:
Originally the functions were static so that they lived long enough for function_handler's reference to them to still be valid when handling a request.  Then they became thread_local so that the variables inside the handler (particularly the auth object) were not getting hit cross-core (auth has config bindings that assert on that).

However, you're right -- we can avoid both qualifiers by just moving the lambdas into the `function_handler` constructor, that way each shard is constructing a fresh one on the heap.  Cleaned that up.